### PR TITLE
Make Control Id Parameters to Constructor

### PR DESCRIPTION
**Pull request by jjtny1**
This PR makes the control_id fields parameters of the constructor. The fields on the contract are immutable so when the contract is deployed, the provided values will be hardcoded into the deployed bytecode.

Closes: https://github.com/risc0/bonsai-foundry-template/issues/46

### CODE REVIEW

File: Cargo.toml
Comment (jjtny1) [inline Cargo.toml:67]: The build is breaking because foundry-cli uses the latest ethers (v2.0.9) which has a dependency conflict with cargo-dependencies uses in cargo-risczero

Screen Shot 2023-08-24 at 7 49 16 PM

Comment (nategraf) [reply Cargo.toml:67]: I think with #810 now merged this should no longer be an issue. Sorry for introducing such a radioactive dependency to the repo 😅

File: bonsai/ethereum/contracts/BonsaiTest.sol
Comment (nategraf) [inline bonsai/ethereum/contracts/BonsaiTest.sol:58]: This factoring makes it impossible to run `forge test` without setting those environment variables. It also does not provide any instructions for how the user should locate these parameters. Are you still working on that? I think we should make sure these are set automatically to a reasonable default before merging this PR

Comment (jjtny1) [reply bonsai/ethereum/contracts/BonsaiTest.sol:58]: Hey yes, im still working on that part. Wanted to see if anyone had a better way of doing thing

File: bonsai/ethereum/contracts/groth16/RiscZeroGroth16Verifier.sol
Comment (nategraf) [reply bonsai/ethereum/contracts/groth16/RiscZeroGroth16Verifier.sol:81]: Can we get these checked in to `risc0`? We have the control IDs for other recursion predicates there, but this predicate seems to me missing. IIRC, this predicate is referred to as `identity_p254`?

**Comment by nategraf** [comment]
This looks good to me overall. Happy to approve when we have some instructions for developers (including ourselves) on how to get these values when they need to be updated.

File: bonsai/ethereum/contracts/groth16/RiscZeroGroth16Verifier.sol
Comment (capossele) [reply bonsai/ethereum/contracts/groth16/RiscZeroGroth16Verifier.sol:81]: #831 should address that

**Comment by capossele** [comment]
#831 merged the ability to get those values by running: `cargo run --bin bonsai-ethereum-contracts -F control-id`. Maybe we can include these instruction on this PR?

**Review by capossele — APPROVED** [review]

**Pull request merged**
